src/app.jsx

```diff
@@ -82,41 +77,37 @@ export const App = () => {
 
     useEvent(superuser, "changed");
     useEffect(() => {
-        Promise.all([initState(), unknownConnectionName()])
-                .then(([_init, connectionNames]) => {
-                    return Promise.allSettled(connectionNames.map(conn => {
-                        return getLibvirtVersion({ connectionName: conn })
-                                .then(() => {
-                                    return getApiData({ connectionName: conn })
-                                            .then(promises => {
-                                                const errorMsgs = [];
-                                                promises.forEach(promise => {
-                                                    if (promise.status == 'rejected' && promise.reason)
-                                                        errorMsgs.push(promise.reason.message);
-                                                });
-                                                setError(errorMsgs.join(', '));
-                                            });
-                                }, ex => {
-                                    // access denied is expected for unprivileged session
-                                    if (conn !== 'system' || superuser.allowed || ex.name !== 'org.freedesktop.DBus.Error.AccessDenied')
-                                        console.error("Failed to get libvirt version from the dbus API:", ex);
-                                    /* If the API call failed on system connection and the user has superuser privileges then show the Empty state screen */
-                                    if (conn == "system")
-                                        setSystemSocketInactive(true);
-                                });
-                    }));
-                })
-                .finally(() => setLoadingResources(false));
+        (async () => {
+            await initState();
+            const connectionNames = await unknownConnectionName();
+
+            await Promise.allSettled(connectionNames.map(async connectionName => {
+                try {
+                    await getLibvirtVersion({ connectionName });
+                    const promises = await getApiData({ connectionName });
+                    const errorMsgs = promises
+                            .filter(promise => promise.status === 'rejected')
+                            .map(promise => promise.reason.message);
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1404-20240129-100518-70521786-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)